java/daikon/chicory/MethodGen24.java

```diff
@@ -0,0 +1,347 @@
+  protected AccessFlags accessFlags;
+
+  /** The method's name. */
+  private String methodName;
+
+  /** The method's type descriptor. */
+  private MethodTypeDesc mtd;
+
+  /** Whether of not the method is static. */
+  private boolean isStatic;
+
+  /** The method's CodeAttribute. */
+  private @Nullable CodeAttribute codeAttribute;
+
+  // fields of the code attribute
+  /** The method's maximum number of locals. */
+  private int maxLocals;
+
+  /** The method's maximum stack size. */
+  private int maxStack;
+
+  /** The name of the method's enclosing class. */
+  private String className;
+
+  /** The method's instruction list. */
+  private List<CodeElement> codeList;
+
+  /** The method's descriptor. */
+  private String descriptor;
+
+  /** The method's signature. */
+  private String signature;
+
+  // fields of the MethodTypeDescriptor
+  /** The method's argument types. */
+  private ClassDesc[] argTypes;
+
+  /** The method's return type. */
+  private ClassDesc returnType;
+
+  /** The method's argument names. */
+  private String[] argNames;
+
+  /** The method's local variable table. */
+  private LocalVariable[] localVariables;
+
+  /**
+   * Constructor for the MethodGen24 object.
+   *
+   * @param methodModel for the method
+   * @param className for the containing class
+   * @param inst_obj the daikon.chicory.Instrument instance
+   */
+  public MethodGen24(
+      final MethodModel methodModel, final String className, daikon.chicory.Instrument24 inst_obj) {
+
+    accessFlags = methodModel.flags();
+    methodName = methodModel.methodName().stringValue();
+    // System.out.println("methodName: " + methodName);
+    descriptor = methodModel.methodType().stringValue();
+    this.className = className;
+    isStatic = accessFlags.has(AccessFlag.STATIC);
+
+    Optional<CodeModel> code = methodModel.code();
+    if (code.isPresent()) {
+      this.code = code.get();
+      // the original elementList is immutable, so we need to make a copy
+      this.codeList = new LinkedList<CodeElement>(this.code.elementList());
+    } else {
+      this.code = null;
+      this.codeList = new ArrayList<>();
+    }
+
+    codeAttribute = methodModel.findAttribute(Attributes.code()).orElse(null);
+    if (codeAttribute != null) {
+      maxLocals = codeAttribute.maxLocals();
+      // It seems like a checker bug that this assert is needed.
+      assert codeAttribute != null
+          : "@AssumeAssertion(nullness): just tested";
+      maxStack = codeAttribute.maxStack();
+    } else {
+      maxLocals = 0;
+      maxStack = 0;
+    }
+
+    SignatureAttribute sa = methodModel.findAttribute(Attributes.signature()).orElse(null);
+    if (sa != null) {
+      signature = sa.signature().stringValue();
+    } else {
+      // if no signature then probably no type arguments so descriptor will do
+      signature = descriptor;
+    }
+
+    mtd = methodModel.methodTypeSymbol();
+    argTypes = mtd.parameterArray();
```

Comment:
The use of both "arg" and "parameter" here indicates that one of the two names is not applicable.